shell/browser/browser_linux.cc

```diff
@@ -139,7 +151,35 @@ bool Browser::SetBadgeCount(std::optional<int> count) {
   }
 }
 
-void Browser::SetLoginItemSettings(LoginItemSettings settings) {}
+void Browser::SetLoginItemSettings(LoginItemSettings settings) {
+  dbus::Bus::Options options;
+  auto bus = base::MakeRefCounted<dbus::Bus>(options);
+
+  dbus::ObjectProxy* object_proxy = bus->GetObjectProxy(
+      kFreedesktopPortalName, dbus::ObjectPath(kFreedesktopPortalPath));
+  dbus::MethodCall method_call(kFreedesktopPortalBackground,
+                               "RequestBackground");
+
+  dbus::MessageWriter writer(&method_call);
+  writer.AppendString("");
+
+  dbus::MessageWriter options_writer(nullptr);
+  writer.OpenArray("{sv}", &options_writer);
+
+  dbus::MessageWriter autostart_writer(nullptr);
+  options_writer.OpenDictEntry(&autostart_writer);
+  autostart_writer.AppendString("autostart");
+  autostart_writer.AppendVariantOfBool(settings.open_at_login);
+  options_writer.CloseContainer(&autostart_writer);
+
+  writer.CloseContainer(&options_writer);
+
+  std::unique_ptr<dbus::Response> response =
+      object_proxy
+          ->CallMethodAndBlock(&method_call,
+                               dbus::ObjectProxy::TIMEOUT_USE_DEFAULT)
+          .value_or(nullptr);
+}
 
 v8::Local<v8::Value> Browser::GetLoginItemSettings(
```

Comment:
We'll also need to make sure that the object returned by `getLoginItemSettings` accurately reflects what the user set in `setLoginItemSettings` - it currently won't.

Reply:
So from the background portal API, we can never get a read-only reading of whether autostart is enabled. Thus, on a cold start of electron, there would be no way of returning a correct value for getLoginItemSettings, without setting the values first and using the response for the request to determine whether the autostart was enabled correctly. From then on we can cache it of-course, but it would only stay around as long as the app is open.

Reply:
What's the planned experience for this feature? Is there no ability to toggle launching on startup? It seems necessary to also have `getLoginItemSettings` implemented for most apps to properly make use of this.

Reply:
I guess it depends on whether the app keeps track of this setting by itself, or whether it relies on the os to do it. Right now, there is no other way to - from within electron - enable autostart in e.g. flatpak. Having just the setting working would already be an improvement, since this would make at least apps that track autostart state themselves work.

Reply:
Would making a request to [org.freedesktop.impl.portal.PermissionStore.Lookup](https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.impl.portal.PermissionStore.html#org-freedesktop-impl-portal-permissionstore-lookup) with the `background` ID happen to return the autostart data?

Reply:
Or possibly sending a request to org.freedesktop.portal.Background.RequestBackground with no options configured?

Reply:
I think given the limitation defined by https://github.com/flatpak/xdg-desktop-portal/issues/1072, I'd be fine with moving forward with only a setter. It seems like future plans may be to have apps display a system panel for toggling autostart.